core/builder/src/main/java/io/quarkus/builder/BuildExecutionBuilder.java

```diff
@@ -24,8 +24,8 @@ public final class BuildExecutionBuilder {
     BuildExecutionBuilder(final BuildChain buildChain, final String buildTargetName) {
         this.buildChain = buildChain;
         this.buildTargetName = buildTargetName;
-        initialSingle = new HashMap<>(buildChain.getInitialSingleCount());
-        initialMulti = new HashMap<>(buildChain.getInitialMultiCount());
+        initialSingle = new HashMap<>();
```

Comment:
Just a question, why not allow use the know capacity?

Reply:
Those 2 variable were initialized to 0 [here](https://github.com/quarkusio/quarkus/pull/36696/files#diff-4c8358bbe5988fdd40e30673a69d4bc524e3ea3e231bfc39e449e444d1d0526eL156), never touched again during the graph linearization algorithm, inserted in the `buildChain` result (still always with value 0) [here](https://github.com/quarkusio/quarkus/pull/36696/files#diff-4c8358bbe5988fdd40e30673a69d4bc524e3ea3e231bfc39e449e444d1d0526eL296) and then read to initialize the size of the maps were you pointed. In essence those maps were **always** initialized with capacity 0.